open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -18,60 +18,59 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
 import java.io.IOException;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.time.Duration;
-import java.time.OffsetDateTime;
 import java.util.ArrayList;
-import java.util.Date;
-import java.util.Map;
 
 /**
  * Class to handle idempotency related operations.
  */
 public class IdempotencyValidator {
 
     private static final Log log = LogFactory.getLog(IdempotencyValidator.class);
-    private static final Map<String, Object> configs = OpenBankingConfigParser.getInstance().getConfiguration();
     private static final ConsentCoreService consentCoreService = ConsentExtensionsDataHolder.getInstance()
             .getConsentCoreService();
 
     /**
      * Method to check whether the request is idempotent.
-     * This method will first check whether idempotency validation is enabled. Then it will check whether the
-     * idempotency key exists in the database and whether the request is received within the allowed time.
+     * This method will first check whether idempotency validation is enabled. After that it will validate whether
+     * required parameters for validation is present.
+     * For validation, need to check whether the idempotency key values is present as a consent attribute, if present
+     * the consent will be retrieved. Finally following conditions will be validated.
+     *  - Whether the client id sent in the request and client id retrieved from the database are equal
+     *  - Whether the difference between two dates is less than the configured time
+     *  - Whether payloads are equal
      *
-     * @param idempotencyKeyName    Idempotency Key Name
-     * @param idempotencyKeyValue    Idempotency Key Value
-     * @param request                Request Payload
+     * @param consentManageData            Consent Manage Data
      * @return  IdempotencyValidationResult
      */
-    public static IdempotencyValidationResult validateIdempotency(String idempotencyKeyName, String idempotencyKeyValue,
-                                                                  String request, String clientId)
-            throws ConsentManagementException {
-
-        if (Boolean.parseBoolean((configs.get(IdempotencyConstants.IDEMPOTENCY_IS_ENABLED)).toString())) {
-            // If idempotency key name, value or request is empty then cannot proceed with idempotency validation
-            if (idempotencyKeyName == null || idempotencyKeyValue == null || request.isEmpty() || clientId == null) {
-                log.debug("Idempotency Key , NameValue, Client ID or Request is empty. Hence cannot proceed with " +
+    public IdempotencyValidationResult validateIdempotency(ConsentManageData consentManageData)
+            throws IdempotencyValidationException {
+
+        if (IdempotencyValidationUtils.isIdempotencyEnabledFromConfig()) {
```

Comment:
Shall we return after this line, if this is not enabled? So that we can improve the indentation for the below lines

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661